_ml-commons-plugin/tutorials/build-chatbot.md

```diff
@@ -0,0 +1,910 @@
+
+Deploy the model:
+
+```json
+POST /_plugins/_ml/models/your_LLM_model_id/_deploy
+```
+{% include copy-curl.html %}
+
+Test the model:
+
+```json
+POST /_plugins/_ml/models/your_LLM_model_id/_predict
+{
+  "parameters": {
+    "prompt": "\n\nHuman: how are you? \n\nAssistant:"
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Step 3: Create an agent with the default prompt
+
+Next, create and test an agent.
+
+### Create an agent
+
+Create an agent of the `conversational` type. 
+
+The agent is configured with the following information:
+
+- Meta information: `name`, `type`, `description`.
+- LLM information: The agent uses an LLM to reason and select the next step, including choosing an appropriate tool and preparing the tool input.
+- Tools: A tool is a function that can be executed by the agent. Each tool can define its own `name`, `description`, and `parameters`.
+- Memory: Stores chat messages. Currently, OpenSearch only supports one memory type: `conversation_index`.
+
+The agent contains the following parameters:
+
+- `conversational`: This agent type has a built-in prompt. To override it with your own prompt, see [Step 4](#step-4-optional-create-an-agent-with-a-custom-prompt).
+- `app_type`: Specify this parameter for reference purposes in order to differentiate between multiple agents.
+- `llm`: Defines the LLM configuration:
+   - `"max_iteration": 5`:  The agent runs the LLM a maximum of five times.
+   - `"response_filter": "$.completion"`: Needed to retrieve the LLM answer from the Bedrock Claude model response.
+   - `"message_history_limit": 5`: The agent retrieves a maximum of the five most recent history messages and adds them to the LLM context. Set this parameter to `0` to omit message history in the context.
+   - `disable_trace`: If `true`, the agent does not store trace data in memory. Trace data is included in each message and provides a detailed recount of steps performed while generating the message.
+- `memory`: Defines how to store messages. Currently, OpenSearch only supports the `conversation_index` memory, which stores messages in a memory index.
+- Tools: 
+   - An LLM will reason to decide which tool to run and will prepare the tool's input. 
+   - To include the tool's output in the response, specify `"include_output_in_agent_response": true`. In this tutorial, you will include the `PPLTool` output in the response (see the example response in [Test the agent](#test-the-agent)). 
+   - By default, the tool's `name` is the same as the tool's `type`, and each tool has a default description. You can override the tool's `name` and `description`.
+   - Each tool in the `tools` list must have a unique name. For example, the following demo agent defines two tools of the `VectorDBTool` type with different names (`population_data_knowledge_base` and `stock_price_data_knowledge_base`). Each tool has a custom description so the LLM can easily understand what the tool does.
```

Comment:
```suggestion
   - Each tool in the `tools` list must have a unique name. For example, the following demo agent defines two tools of the `VectorDBTool` type with different names (`population_data_knowledge_base` and `stock_price_data_knowledge_base`). Each tool has a custom description so that the LLM can easily understand what the tool does.
```